### ISSUE TITLE
Excalidraw Mode not creating proper Dataview Fields

### ISSUE THREAD

**Issue by 1activegeek** [zsviczian/obsidian-excalidraw-plugin#691]
**Describe the bug**
I'm new to Excalidraw so perhaps I'm missing something glaring here, but I can't seem to find a proper way to be able to simply enter text into my drawing in the format of `Field:: [[Value]]` and have Dataview manage to pick it up. And I've noticed that if I try to enter Markdown mode to enter the same text, but I have it in the drawing, it still won't work. However if I remove it from the drawing, the Markdown text does work. 

**To Reproduce**
1. Create new Excalidraw document
2. Tap the Text button and enter text such as `Test:: [[Blue]]`
3. Check Markdown mode to ensure the text is showing properly (it should)
4. Create a new document and create a new data view query such as below. 
````
```dataview
LIST
WHERE Test = [[Blue]]
```
````
5. Check that the resulting Query shows 0 results6. 
6. Modify the Excalidraw file from step 1, and add the same text in the markdown view7. 
7. Check the document from Step 4, and notice there is still no result from the query8. 
8. Once again edit the Excalidraw document, removing the text from the Excalidraw view now raw Markdown9. 
9. Check the Query in the new document, and see that it will now show this Excalidraw file in the list10. 

**Expected behavior**
I'm expecting that the behavior seen by simply entering the Markdown text, but done in Excalidraw view, should render a result in the Dataview query being used.

**Screenshots**
The Query being used
<img width="590" alt="image" src="https://user-images.githubusercontent.com/6810720/178072615-a4fe367e-1088-4cf9-ac79-c1657dea6844.png">

The Field being noted in the Excalidraw document
<img width="810" alt="image" src="https://user-images.githubusercontent.com/6810720/178072709-f09b25ea-ec5a-41ff-8980-ca96f8a2793a.png">

A view of the file in Markdown mode , entering in the field, no results still
<img width="635" alt="image" src="https://user-images.githubusercontent.com/6810720/178072843-240dc005-d773-43db-a20f-2326afab191c.png">

Finally removing the field from the Excalidraw view
<img width="955" alt="image" src="https://user-images.githubusercontent.com/6810720/178072871-1fe7c2a7-4fd1-49a2-bf50-f8d20cf65f0d.png">

Resulting in the actual result showing in the query used from the first screenshot
<img width="730" alt="image" src="https://user-images.githubusercontent.com/6810720/178072904-4b0734b4-e2bb-4934-b1e8-81d15e770b6d.png">

**Environment (please complete the following information):**
 - OS including version: macOS Monterey 12.4
 - Plugin version: 1.6.34
 - Obsidian version: 0.14.15

**Additional context**
Basically seeing the same behavior (I think) as seen in #633, but not relevant to Excalibrain and makes no difference if the Compress option is on. I have it on, and want it on. But I've not seen anything even change in the file itself with that on. I don't believe it has anything to do with the link either, as I've tried just raw text, and same results.

**Comment by zsviczian** [comment]
The short answer is to add an empty line after `Test:: [[Blue]]`
If you check the markdown file, the block reference for the field is also part of the `Test:: ` field's value.

https://user-images.githubusercontent.com/14358394/178150914-8a2d439f-402c-4d12-8764-d301ebab21c2.mp4

An alternative solution is to use the mid-sentence field format... You can either use parenthesis or brackets.

https://user-images.githubusercontent.com/14358394/178151033-11a7687f-6c1a-417e-a5c1-47c6f896a197.mp4

**Comment by 1activegeek** [comment]
This is fantastic - obviously I'd love prettier options, but now that I see you illustrating it I hadn't even stopped to realize the block reference is part of the actual field value ... Duh! That perfectly explains the core issue. I also was not aware I could use the brackets as you indicated in the second solution - new Obsidian learning!! 

Thank you so much for taking the time to not only answer the question, help illustrate it, and highlight the importance bit I was missing all in the same shot. Appreciate your time and work on this plugin - hopefully now I can really start to dig into using this more when taking handwritten notes for my customers and having Dataview pull it together in perfect views!
